datastore/protocol.go

```diff
@@ -0,0 +1,281 @@
+package datastore
+
+import (
+	"context"
+	"slices"
+	"strings"
+
+	nex "github.com/PretendoNetwork/nex-go"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/globals"
+	datastore "github.com/PretendoNetwork/nex-protocols-go/datastore"
+	datastore_super_mario_maker "github.com/PretendoNetwork/nex-protocols-go/datastore/super-mario-maker"
+	datastore_types "github.com/PretendoNetwork/nex-protocols-go/datastore/types"
+	"github.com/minio/minio-go/v7"
+)
+
+var commonDataStoreProtocol *CommonDataStoreProtocol
+
+type CommonDataStoreProtocol struct {
+	server                  *nex.Server
+	DefaultProtocol         *datastore.Protocol
+	SuperMarioMakerProtocol *datastore_super_mario_maker.Protocol
+
+	s3Bucket                                            string
+	s3DataKeyBase                                       string
+	s3NotifyKeyBase                                     string
+	rootCACert                                          []byte
+	MinIOClient                                         *minio.Client
+	S3Presigner                                         S3PresignerInterface
```

Comment:
do we have to export these?

Reply:
These are exported for if a developer wants to use them without having to define them manually in each server. For example in the SMM server there's multiple methods which need presigned URLs so it can just use the exported `S3Presigner` here and not make one itself. Though yeah the `MinIOClient` is less useful exported since the server needs to make it already